examples/server/server.cpp

```diff
@@ -2297,7 +2297,7 @@ struct server_context {
             for (size_t i = 0; i < std::min(n_vocab, n_probs); i++) {
                 result.probs.push_back({
                     cur[i].id,
-                    common_detokenize(ctx, {cur[i].id}, special),
+                    common_detokenize(ctx, {cur[i].id}, special, /* remove_space_prefix */ false),
```

Comment:
Wouldn't it be simpler to detokenize `{bos, cur[i].id}`?

I would like t avoid adding this extra parameter to the API.

Reply:
@ggerganov maybe I misunderstood, but won't `{bos, cur[i].id}` returns something like `<bos> text`? But what we want here is just ` text`

Also would be nicer if we can query the raw token from vocab, but I don't think we have an API for that.

Reply:
If `special == false` it won't render the BOS token.

Reply:
> Also would be nicer if we can query the raw token from vocab, but I don't think we have an API for that.

What about `llama_token_to_piece`?

Reply:
@ggerganov @ngxson Thanks, that is a much simpler and better solution!